### PR TITLE
feat: adds edit and delete functionality to userlist

### PR DESCRIPTION
**Pull request by derrabauke**
-  edit & delete buttons in userlist
 - adds **global** confirmation dialog when deleting resources via `EditForm>>delete`

### CODE REVIEW

File: addon/decorators/confirm-task.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  if (args.length === 3 && !args[0].message) {
+    // We can assume that the decorator was called without options
+    return decorateFn(...args);
+  }
+
+  return decorateFn;
+};
+
+export function confirmTask(...decoratorArgs) {
+  const options = validate(decoratorArgs) ? decoratorArgs[0] : {};
```
Comment (czosel) [inline addon/decorators/confirm-task.js:83]:
```suggestion
  const options = initOptions(decoratorArgs);
```

File: addon/decorators/confirm-task.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+          ...translatedOptions,
+          ...filteredOptions,
+        }))
+      ) {
+        return;
+      }
+      return yield* gen.apply(this, args);
+    };
+  }
+  return makeFlexibleDecorator(decorate, decoratorArgs);
```
[the remark this answers is not in the document]
Comment (derrabauke) [reply addon/decorators/confirm-task.js:111]: I thought of a default case, where the decorator is used without any arguments. Had the generalization in mind. Maybe a thought for the future implementation in ember-uikit.

**Review by czosel — APPROVED** [review]
:+1:

**Pull request merged**